default/AI/DiplomaticCorp.py

```diff
@@ -98,7 +92,8 @@ def evaluate_diplomatic_attitude(self, other_empire_id):
         num_peace_requests = len(foAI.foAIstate.diplomatic_logs.get('peace_requests', {}).get(log_index, []))
         num_war_declarations = len(foAI.foAIstate.diplomatic_logs.get('war_declarations', {}).get(log_index, []))
         # Too many requests for peace irritate the AI, as do any war declarations
-        irritation = (foAI.foAIstate.aggression * (2.0 + num_peace_requests / 10.0 + 2.0 * num_war_declarations) + 0.5)
+        aggression = foAI.foAIstate.character.get_behavior(Aggression)
```

Comment:
You are right.  That needs to be abstracted into a Behavioral question.  I remember looking at it and it was so specific that I couldn't think of an abstracted behavioral question that encompassed this equation.  So, I left it for later and later became never.  I will fix that.